core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/AppSQLPlanAnalyzer.scala

```diff
@@ -78,6 +79,29 @@ class AppSQLPlanAnalyzer(app: AppBase, appIndex: Int) extends AppAnalysisBase(ap
   val IODiagnosticMetricsMap: HashMap[(Long, Long), ArrayBuffer[SQLAccumProfileResults]] =
     HashMap.empty[(Long, Long), ArrayBuffer[SQLAccumProfileResults]]
 
+  // A list of AccumProfileResults objects containing filtered diagnostic metrics for the top 7
+  // stages with the highest durations.
+  val filteredStageAccumList: ArrayBuffer[AccumProfileResults] = ArrayBuffer[AccumProfileResults]()
+
+  // A mapping from accumulator ID to a tuple of (SQL ID, Node ID, Node Name).
+  val accumIdToNodeInfoMap: HashMap[Long, (Long, Long, String)] =
+    HashMap.empty[Long, (Long, Long, String)]
+
+  /**
+   * Retrieves the top 7 stage IDs with the highest execution durations.
+   * 7 is a magic number that was chosen from past experiences.
+   */
+  def topDurationStageIds: Set[Int] = {
```

Comment:
Are we querying this set multiple times? Should we make this a lazy val?

Reply:
thanks @parthosa, updated this.